src/app/pages/Collectibles/components/ModelViewer.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC, useEffect, useRef } from 'react';
+
+import '@google/model-viewer';
+import ModelViewerElementBase from '@google/model-viewer/lib/model-viewer-base';
+import classNames from 'clsx';
+
+import { emptyFn } from 'app/utils/function.utils';
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'model-viewer': React.DetailedHTMLProps<
+        React.AllHTMLAttributes<Partial<globalThis.HTMLElementTagNameMap['model-viewer']>>,
+        Partial<globalThis.HTMLElementTagNameMap['model-viewer']>
+      >;
+    }
+  }
+}
+
+interface Props {
+  uri: string;
+  alt?: string;
+  className?: string;
+  onError?: EmptyFn;
+}
+
+export const ModelViewer: FC<Props> = ({ uri, alt, className, onError = emptyFn }) => {
+  const modelViewerRef = useRef<ModelViewerElementBase>(null);
+
+  useEffect(() => {
+    const modelViewer = modelViewerRef.current;
+
+    if (modelViewer) {
+      modelViewer?.addEventListener('error', onError);
+
+      return () => modelViewer?.removeEventListener('error', onError);
+    }
+
+    return undefined;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [modelViewerRef.current]);
```

Comment:
As far as I know, this dependency has no effect because it comes from a ref object. Please try removing it and checking how the component works.

Reply:
indeed